tests/playwright-test/playwright.trace.spec.ts

```diff
@@ -566,7 +566,7 @@ test('should opt out of attachments', async ({ runInlineTest, server }, testInfo
     'After Hooks',
   ]);
   expect(trace.actions[1].attachments).toEqual(undefined);
-  expect([...trace.resources.keys()].filter(f => f.startsWith('resources/'))).toHaveLength(0);
+  expect([...trace.resources.keys()].filter(f => f.startsWith('resources/'))).toEqual([expect.stringMatching(/^resources\/src@.*$/)]);
```

Comment:
I wonder how did this test work before? This PR does not seem to affect this behavior, does it?

Reply:
I'm not sure why exactly, but some change in this PR made [`filteredStack`](https://github.com/Skn0tt/playwright/blob/365b30f64ffb9c113b92c337d8b5d54c80096c99/packages/playwright/src/worker/testInfo.ts#L263) contain something when `testInfo.attach()` is called. That leads tracing to include the source code as a resource. Let me refactor this assertion to make that more clear.

Reply:
there you go: https://github.com/microsoft/playwright/pull/34037/commits/d27755117742837822c8558cc741c875624a67c1